File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/model/LogIconColumnRenderer.java
package ru.kmorozov.gbd.logger.model;

import javax.swing.*;
import javax.swing.table.DefaultTableCellRenderer;
import java.awt.*;
import java.util.logging.Level;

/**
 * Created by km on 18.12.2015.
 */
public class LogIconColumnRenderer extends DefaultTableCellRenderer {

    private final JLabel labelInfo = new JLabel(null, new ImageIcon(this.getClass().getResource("/images/icons/info.png")), SwingConstants.CENTER);
    private final JLabel labelSevere = new JLabel(null, new ImageIcon(this.getClass().getResource("/images/icons/error.png")), SwingConstants.CENTER);
    private final JLabel labelFinest = new JLabel(null, new ImageIcon(this.getClass().getResource("/images/icons/warning.png")), SwingConstants.CENTER);

    @Override
    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
        Level level = (Level) value;

        if (level == Level.INFO) return this.labelInfo;
        else if (level == Level.SEVERE) return this.labelSevere;
        else return this.labelFinest;
    }
}

File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/model/LogIconColumnRenderer.kt
package ru.kmorozov.gbd.logger.model

import javax.swing.*
import javax.swing.table.DefaultTableCellRenderer
import java.awt.*
import java.util.logging.Level

/**
 * Created by km on 18.12.2015.
 */
class LogIconColumnRenderer : DefaultTableCellRenderer() {

    private val labelInfo = JLabel(null, ImageIcon(javaClass.getResource("/images/icons/info.png")), SwingConstants.CENTER)
    private val labelSevere = JLabel(null, ImageIcon(javaClass.getResource("/images/icons/error.png")), SwingConstants.CENTER)
    private val labelFinest = JLabel(null, ImageIcon(javaClass.getResource("/images/icons/warning.png")), SwingConstants.CENTER)

    override fun getTableCellRendererComponent(table: JTable?, value: Any, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component {
        val level = value as Level

        return if (level === Level.INFO)
            labelInfo
        else if (level === Level.SEVERE)
            labelSevere
        else
            labelFinest
    }
}
